### PR TITLE
Default BBox returning changed for someitems

### PR DESCRIPTION
**Pull request by maxGimeno**
- Using tools like the selection tool creates items that make the scene
  re-center, and this can be annoying. To avoid that, I changed the default BBox
  of those items from BBox(), which is (0,0,0,1,1,1) to scene->bbox() for
  most of the cases. This way the scene bbox does not change and you don't
  lose your position when they are added.

### CODE REVIEW

**Comment by sloriot** [comment]
+1

**Comment by sloriot** [comment]
Merged in [CGAL-4.8-Ic-17](https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.8-Ic-17.shtml)

**Comment by lrineau** [comment]
If I remember well, that type of issue should be solved by using those functions of `Scene_item`:
```C++
  virtual bool isFinite() const { return true; }
  virtual bool isEmpty() const { return true; }
```

File: Polyhedron/demo/Polyhedron/Scene.cpp [hunk 1]
```diff
@@ -58,12 +58,14 @@ Scene::addItem(Scene_item* item)
     connect(item, SIGNAL(renderingModeChanged()),
             this, SLOT(callDraw()));
     if(bbox_before + item->bbox() != bbox_before)
-{ Q_EMIT updated_bbox(); }
+    {
+        Q_EMIT updated_bbox();
+    }
     QAbstractListModel::beginResetModel();
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Scene.cpp]: Here the code should verify `item->isEmpty()` and `item->ifFinite()`.

File: Polyhedron/demo/Polyhedron/Scene.cpp [hunk 2]
```diff
@@ -57,13 +57,17 @@ Scene::addItem(Scene_item* item)
             this, SLOT(itemChanged()));
     connect(item, SIGNAL(renderingModeChanged()),
             this, SLOT(callDraw()));
-    if(bbox_before + item->bbox() != bbox_before)
-{ Q_EMIT updated_bbox(); }
+    if(bbox_before + item->bbox() != bbox_before
+            && item->isFinite()
+            && !item->isEmpty())
+    {
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/Scene.cpp]: `isFinite` and `!isEmpty()` should be tested first.

**Comment by lrineau** [comment]
Will be tested in [CGAL-4.8-Ic-19][t].

[t]: https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.8-Ic-19.shtml

**Comment by maxGimeno** [comment]
I didn't see any error/warning that could have been caused by this PR, I think it can be merged.

**Pull request merged**